markets/storageadapter/dealpublisher.go

```diff
@@ -172,10 +187,17 @@ func (p *DealPublisher) ForcePublishPendingDeals() {
 }
 
 func (p *DealPublisher) Publish(ctx context.Context, deal market.ClientDealProposal) (cid.Cid, error) {
-	pdeal := newPendingDeal(ctx, deal)
+	pdeal, err := newPendingDeal(ctx, deal)
+	if err != nil {
+		return cid.Undef, fmt.Errorf("failed to generate cid for the deal proposal: %w", err)
+	}
 
 	// Add the deal to the queue
-	p.processNewDeal(pdeal)
+	if p.manualPSD {
+		p.queueDeals(pdeal)
+	} else {
+		p.processNewDeal(pdeal)
+	}
```

Comment:
I think we can simplify by just making some modifications to processNewDeal, rather than adding a new queueDeals method: in processNewDeal we should return after adding the deal to the queue if p.manualPSD is set